### PR TITLE
support av1 low overhead obu demux

### PR DESCRIPTION
**Pull request by xuguangxin**
it's used by netflix http://download.opencontent.netflix.com/?prefix=AV1/Chimera/

### CODE REVIEW

**Comment by xuguangxin** [comment]
@feiwan1 @xhaihao @guoyejun ,
please help review.
thanks

File: libavcodec/av1_parse.h [hunk 1]
```diff
@@ -99,10 +99,10 @@ static inline int64_t leb128(GetBitContext *gb) {
 
 static inline int parse_obu_header(const uint8_t *buf, int buf_size,
                                    int64_t *obu_size, int *start_pos, int *type,
-                                   int *temporal_id, int *spatial_id)
+                                   int *temporal_id, int *spatial_id, int* has_size_flag)
```
Comment (xhaihao) [inline libavcodec/av1_parse.h]: It'd be better to keep the coding style unchanged.  s/int*/int *

File: libavformat/av1dec.c [hunk 2]
```diff
@@ -57,17 +60,17 @@ static int leb(AVIOContext *pb, uint32_t *len) {
     return i;
 }
 
-static int read_obu(const uint8_t *buf, int size, int64_t *obu_size, int *type)
+static int read_obu(const uint8_t *buf, int size, int64_t *obu_size, int *type, int* has_size_flag)
```
Comment (xhaihao) [inline libavformat/av1dec.c]: s/int*/int *

File: libavformat/av1dec.c [hunk 3]
```diff
@@ -57,17 +60,17 @@ static int leb(AVIOContext *pb, uint32_t *len) {
...
     int len;
 
     len = parse_obu_header(buf, size, obu_size, &start_pos,
-                           type, &temporal_id, &spatial_id);
+                           type, &temporal_id, &spatial_id, has_size_flag);
     if (len < 0)
         return len;
 
-    return 0;
+    return len;
```
Comment (xhaihao) [inline libavformat/av1dec.c]: so we may remove the if statement in line #70-71, right?

File: libavformat/av1dec.c [hunk 4]
```diff
@@ -137,16 +140,15 @@ static int annexb_probe(const AVProbeData *p)
...
+        temporal_unit_size -= obu_unit_size;
+        frame_unit_size -= obu_unit_size;
     } while (frame_unit_size);
 
     return 0;
 }
 
-static int annexb_read_header(AVFormatContext *s)
+static int read_header(AVFormatContext *s, AVRational framerate, AVBSFContext** bsf, void* c)
 {
```
Comment (xhaihao) [inline libavformat/av1dec.c:146]: s/AVBSFContext**/AVBSFContext **
s/void*/void *

File: libavformat/av1dec.c [hunk 5]
```diff
@@ -251,12 +260,158 @@ static int annexb_read_close(AVFormatContext *s)
...
+
+        left = c->len - size;
+        memcpy(c->prefetched, c->prefetched + size, left);
+        c->len = left;
+    } else {
+        memcpy(pkt->data, c->prefetched, c->len);
+        left = size - c->len;
+        ret = avio_read(s->pb, pkt->data + c->len, left);
+        if (ret != left) {
+            av_log(c, AV_LOG_ERROR, "Failed to read %d frome file\n", left);
```
Comment (xhaihao) [inline libavformat/av1dec.c:333]: Need to free the allocated pkt?

Comment (xuguangxin) [reply libavformat/av1dec.c:333]: no, we do not need.  ff_read_packet will unref the packet

https://github.com/intel-media-ci/ffmpeg-cartwheel/blob/c931ca68b806e7204186aca9ffbb7bfb8988437a/libavformat/utils.c#L851-L853

**Comment by xuguangxin** [comment]
@xhaihao done

**Comment by xuguangxin** [comment]
close this since it already send to upstream for review. 
https://patchwork.ffmpeg.org/project/ffmpeg/list/?series=1965
thanks haihao for suggestions

**Pull request closed without merging**